webpack/scenes/Hosts/ChangeContentSource/components/Hosts.js

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from 'react';
+import {
+  Button,
+  GridItem,
+  Label,
+  Modal,
+  ModalVariant,
+  List, ListItem,
+  SearchInput,
+} from '@patternfly/react-core';
+
+import { translate as __ } from 'foremanReact/common/I18n';
+import { foremanUrl } from 'foremanReact/common/helpers';
+
+import PropTypes from 'prop-types';
+
+const Hosts = ({
+  contentHosts, hostsWithoutContent,
+}) => {
+  const [modal, handleModal] = useState(false);
+  const [modalHosts, setModalHosts] = useState([]);
+  const [modalTitle, setModalTitle] = useState('Hosts');
+  const [search, setSearch] = useState('');
+
+  const loadModal = (items, title) => {
```

Comment:
It's generally not a good idea to store values in state that can be computed or derived from other data. It looks like we're using the `loadModal` function below to decide if this should be a "Hosts to update" or "Ignored hosts" modal. Instead, I think the code will be much cleaner if you move the `contentHosts` and `hostsWithoutContent` logic to the parent component.  There, you can render a `<Hosts>` component (or two if necessary) and decide _there_ what props to pass to it. This will be much cleaner than using the local state here.